crates/burn-train/src/tester/base.rs

```diff
@@ -0,0 +1,31 @@
+use crate::checkpoint::{FileCheckpointer, MetricCheckpointingStrategy};
+use crate::components::LearnerComponentsMarker;
+use crate::components_test::TesterComponents;
+use crate::Learner;
+use burn_core::optim::adaptor::OptimizerAdaptor;
+use burn_core::optim::Adam;
+use burn_core::record::{DefaultFileRecorder, FullPrecisionSettings};
+use burn_core::tensor::backend::AutodiffBackend;
+
+/// Learner struct encapsulating all components necessary to train a Neural Network model.
+///
+/// To create a learner, use the [builder](crate::tester::LearnerBuilder) struct.
+pub struct Tester<TC: TesterComponents> {
+    pub(crate) learner: Learner<
+        LearnerComponentsMarker<
```

Comment:
Dummy types for what's not needed - for now the first/simplest matching types I found. Could be improved by implementing no-op types / phantom data.